pywt/_extensions/_dwt.pyx

```diff
@@ -98,9 +98,9 @@ cpdef dwt_axis(np.ndarray data, Wavelet wavelet, MODE mode, unsigned int axis=0)
     return (cA, cD)
 
 
-cpdef idwt_single(data_t[::1] cA, data_t[::1] cD, Wavelet wavelet, MODE mode):
-    cdef data_t[::1] rec
+cpdef idwt_single(np.ndarray cA, np.ndarray cD, Wavelet wavelet, MODE mode):
```

Comment:
I tried just using a memoryview for the inputs to `idwt_single`, but leaving the result as np.ndarray, but this did negated a large portion of the speedup (final commit of the plot below):

**For `time_idwt`**
![memoryview_revert](https://cloud.githubusercontent.com/assets/6528957/13618856/4573a602-e554-11e5-8ced-dd96ba256bad.png)